game/entities.py

```
@@ -0,0 +1,252 @@
+                target_height = int(zone_height * (1 - t) + self.heights[start - smooth_distance - 1] * t)
+                self.heights[start - i] = target_height
+        
+        # Smooth transition to the right
+        for i in range(1, smooth_distance + 1):
+            if end + i < self.width:
+                # Linear interpolation
+                t = i / (smooth_distance + 1)
+                target_height = int(zone_height * (1 - t) + self.heights[end + smooth_distance + 1] * t)
+                self.heights[end + i] = target_height
+    
+    def is_point_colliding(self, x, y):
```

Ruff (0.8.2)

Missing return type annotation for public function `is_point_colliding`

(ANN201)